render/render.go

```diff
@@ -189,20 +189,30 @@ func newJoinResults() joinResults {
 }
 
 // Add Node M under id, creating a new result node if not already there
-// and updating the mapping from old ID to new ID
-// Note we do not update any counters for child topologies here, because addToResults
-// is only ever called when m is an endpoint and we never look at endpoint counts
-func (ret *joinResults) addToResults(m report.Node, id string, create func(string) report.Node) {
+// incrementing a counter if nonblank, and updating the mapping from old ID to new ID
+func (ret *joinResults) addToResults(m report.Node, id string, addChildren bool, create func(string) report.Node, counters ...string) {
```

Comment:
This function is doing too much, and i miss the comment about endpoints for the original.

Suggestion:
- rename the original `addToResults` to `addToResultsFromEndpoint` (better naming suggestions welcome)
- add a new `addToResults` that doesn't add children and increments the `m.Topology` counter of the result.

This does entail duplicating seven lines of code.